### PR TITLE
Allow inlineSvg function to take attributes

### PR DESCRIPTION
**Pull request by sammorrisdesign-zz**
Then use that to add titles back to the weather icons

### CODE REVIEW

**Comment by sndrs** [comment]
you need to add the same functionality to the js method too - give me a shout if you need a hand

File: common/app/views/fragments/inlineSvg.scala.html [hunk 1]
```diff
@@ -1,4 +1,4 @@
-@(file: String, path: String = "", classes: Seq[String] = List())
-<span class="inline-@file @if(path){inline-@path} @classes.mkString(" ")">
+@(file: String, path: String = "", classes: Seq[String] = List(), attributes: Map[String, String] = Map())
```
Comment (gklopper) [inline common/app/views/fragments/inlineSvg.scala.html]: What attributes other than `title` are we expecting? Maybe we could just use `title: Option[String] = None`.

Map params get messy real quick.

Comment (sammorrisdesign) [inline common/app/views/fragments/inlineSvg.scala.html]: Originally I was going to do that but you never know when we might want to use a `data-` attribute or `aria`. So I went with this instead as it's a little more future proof.

Comment (sndrs) [inline common/app/views/fragments/inlineSvg.scala.html]: it feels like there's a good argument for just allowing a string for further classes/attribtues. makes it clear to read and seems far less engineered?

Comment (johnduffell) [inline common/app/views/fragments/inlineSvg.scala.html]: if we need more in future we can always refactor it again at that point.  The good thing about real parameters is you get all the compiler help, and we may never need to extend it

**Comment by sndrs** [comment]
also, just added this https://github.com/guardian/frontend/blob/master/docs/inline-svgs.md - need to update that to reflect it too

**Comment by sndrs** [comment]
also, just added this https://github.com/guardian/frontend/blob/master/docs/inline-svgs.md - need to update that to reflect it too

**Comment by sammorrisdesign** [comment]
Added functionality to js and updated documentation. Should be good barring any additional comments.

File: common/app/views/fragments/inlineSvg.scala.html [hunk 2]
```diff
@@ -1,4 +1,4 @@
-@(file: String, path: String = "", classes: Seq[String] = List())
-<span class="inline-@file @if(path){inline-@path} @classes.mkString(" ")">
+@(file: String, path: String = "", classes: Seq[String] = List(), title: Option[String] = None)
+<span class="inline-@file @if(path){inline-@path} @classes.mkString(" ")" title="@title">
```
Comment (sndrs) [inline common/app/views/fragments/inlineSvg.scala.html]: this shoudl be in a conditional, like path - i.e. if none is supplied, don't render it

File: docs/inline-svgs.md [hunk 3]
```diff
@@ -96,7 +98,7 @@ The result of inserting an SVG with either method is the compressed source wrapp
 So both this Scala:
 
 ```scala
-@fragments.inlineSvg("profile-36", "icon", List("rounded-icon", "control__icon-wrapper"))
+@fragments.inlineSvg("profile-36", "icon", List("rounded-icon", "control__icon-wrapper"), Some("Profile icon"))
```
Comment (sndrs) [inline docs/inline-svgs.md]: shoudl this be a `some`? (i don't know, it looks weird to me, but I really don't know)

**Comment by gklopper** [comment]
15 days...

**Comment by phamann** [comment]
16 days...

**Comment by sammorrisdesign** [comment]
Sorry guys. I'm off this week so doubt I'll get time to finish this before we get to the 20s (days, not 2020s).

Unless someone else wants to...

**Comment by phamann** [comment]
21 days. Ship or close.

File: common/app/views/fragments/inlineSvg.scala.html (hunk 2 above)
Comment (robertberry) [inline common/app/views/fragments/inlineSvg.scala.html]:
```scala
@title.map { t => title="@t" }
```

File: docs/inline-svgs.md (hunk 3 above)
Comment (robertberry) [inline docs/inline-svgs.md]: Yeah it should

**Comment by robertberry** [comment]
Apart from my comment :+1:

**Comment by sammorrisdesign** [comment]
We can all move on with our lives..

**Pull request merged**